examples/inspection/plot_model_inspection_cross_validation.py

```diff
@@ -0,0 +1,235 @@
+#
+# Indeed, if the parameters `alpha` are varying depending of the input data, it
+# might be challenging to put our model in production because we will not be
+# able to fix this hyperparameter.
+#
+# Let's check the `alpha` parameter variance.
+
+# %%
+alpha = [est[-1].alpha_ for est in cv_estimators]
+plt.hist(alpha, bins=30, density=True)
+plt.xlabel("Alpha")
+plt.ylabel("Density")
+_ = plt.title("Distribution of alpha parameter \nduring cross-validation")
+
+# %%
+# We see that the regularization parameter `alpha` values are centered and
+# condensed around 40: it is a good sign. It means that most of the models
+# tuned within the cross-validation had an equivalent parameter `alpha`.
+#
+# However, not only hyperparameter such as `alpha` should be studied. The model
+# parameter coming out of the fitting process should analyzed. In our case, we
+# used a linear model. These models are parametrized and rely on two
+# parameters: `coef_` and `intercept_`. Therefore, we should analyze the
+# variance of these parameters as well.
+#
+# For the sake of simplicity, we are going to solely look at the `coef_`.
+# %%
+import pandas as pd
+
+coefficients = pd.DataFrame(
+    [est[-1].coef_ for est in cv_estimators],
+    columns=X.columns,
+)
+coefficients
+
+# %%
+import seaborn as sns
+
+plt.figure(figsize=(9, 7))
+sns.swarmplot(data=coefficients, orient="h", color="k", alpha=0.5)
+sns.boxplot(data=coefficients, orient="h")
+plt.axvline(x=0, color=".5")
+plt.title("Coefficient variability")
+_ = plt.subplots_adjust(left=0.3)
+
+# %%
+# We observe that the coefficients do not vary meaning that all models trained
+# are similar. Each individual model is expected to more or less give the same
+# predictions.
+#
+# Put a predictive model in production
+# ------------------------------------
+#
+# With the above analysis, we can safely create a model by fixing the `alpha`
+# hyperparameter. Subsequently, we can train the model on the full training
+# set.
+
+# %%
+from sklearn.linear_model import Ridge
+
+production_model = make_pipeline(StandardScaler(), Ridge(alpha=40))
+production_model.fit(X_train, y_train)
+
+# %%
+# At the beginning of the process, we left out some data. Now, we can use it
+# to further check if the model performance is similar to what we could expect
+# from the analysis done within the cross-validation framework.
+
+# %%
+print(
+    f"The performance of our production model: "
+    f"R2={production_model.score(X_test, y_test):.2f}"
+)
+
+# %%
+# We see that the performance are comparable to the above performance which is
+# not surprising. Similarly, we could look at the coefficients of the
+# production model and compare it with the coefficients obtained within the
+# cross-validation study.
+#
+# However, you should be aware that this latest step does not give any
+# information about the variance of the model. It should never be used to
+# evaluate the model itself.
```

Comment:
We could add a reference to the nice new example: https://scikit-learn.org/dev/auto_examples/model_selection/plot_grid_search_stats.html#sphx-glr-auto-examples-model-selection-plot-grid-search-stats-py